src/rpc/RPCHelpers.cpp

```diff
@@ -189,13 +189,209 @@ toJson(ripple::STBase const& obj)
+        if (!transaction)
+            return false;
+
+        ripple::TxType const tt = transaction->getTxnType();
+        if (tt != ripple::ttNFTOKEN_MINT && tt != ripple::ttNFTOKEN_ACCEPT_OFFER &&
+            tt != ripple::ttNFTOKEN_CANCEL_OFFER)
+            return false;
+
+        // if the transaction failed nothing could have been delivered.
+        if (transactionMeta.getResultTER() != ripple::tesSUCCESS)
+            return false;
+
+        return true;
+    };
+
+    if (!canHaveNFTokenID())
+        return;
+
+    if (auto const type = transaction->getTxnType(); type == ripple::ttNFTOKEN_MINT)
+    {
+        std::optional<ripple::uint256> result = getNFTokenIDFromPage(transactionMeta);
+        if (result.has_value())
+            response["nftoken_id"] = ripple::to_string(result.value());
+    }
+    else if (type == ripple::ttNFTOKEN_ACCEPT_OFFER)
+    {
+        std::vector<ripple::uint256> result = getNFTokenIDFromDeletedOffer(transactionMeta);
+
+        if (result.size() > 0)
+            response["nftoken_id"] = ripple::to_string(result.front());
+    }
+    else if (type == ripple::ttNFTOKEN_CANCEL_OFFER)
+    {
+        std::vector<ripple::uint256> result = getNFTokenIDFromDeletedOffer(transactionMeta);
+
+        response["nftoken_ids"] = boost::json::array();
+        for (auto const& nftID : result)
+            response["nftoken_ids"].as_array().push_back(boost::json::value(ripple::to_string(nftID)));
+    }
+}
+
+std::optional<ripple::uint256>
+getOfferIDFromCreatedOffer(ripple::TxMeta const& transactionMeta)
+{
+    for (ripple::STObject const& node : transactionMeta.getNodes())
+    {
+        if (node.getFieldU16(ripple::sfLedgerEntryType) != ripple::ltNFTOKEN_OFFER ||
+            node.getFName() != ripple::sfCreatedNode)
+            continue;
+
+        return node.getFieldH256(ripple::sfLedgerIndex);
+    }
+    return std::nullopt;
+}
+
+void
+insertNFTokenOfferID(
+    boost::json::object& response,
+    std::shared_ptr<ripple::STTx const> const& transaction,
+    ripple::TxMeta const& transactionMeta)
+{
+    auto canHaveNFTokenOfferID = [&]() {
+        if (!transaction)
+            return false;
+
+        ripple::TxType const tt = transaction->getTxnType();
+        if (tt != ripple::ttNFTOKEN_CREATE_OFFER)
+            return false;
+
+        // if the transaction failed nothing could have been delivered.
+        if (transactionMeta.getResultTER() != ripple::tesSUCCESS)
+            return false;
+
+        return true;
+    };
+
+    if (!canHaveNFTokenOfferID())
+        return;
+
+    std::optional<ripple::uint256> result = getOfferIDFromCreatedOffer(transactionMeta);
+
+    if (result.has_value())
+        response["offer_id"] = ripple::to_string(result.value());
+}
+
 std::pair<boost::json::object, boost::json::object>
-toExpandedJson(Backend::TransactionAndMetadata const& blobs)
+toExpandedJson(Backend::TransactionAndMetadata const& blobs, NFTokenJSON NFTEnabled)
```

Comment:
This function is called by other handlers, only tx and account_tx can append nftid